app/bundles/CampaignBundle/Membership/MembershipManager.php

```diff
@@ -249,4 +264,17 @@ private function advanceProgressBar()
             $this->progressBar->advance();
         }
     }
+
+    /**
+     * Transform an Exception to an array to pass to the logger.
+     */
+    private function getExceptionAsArray(\Exception $exception): array
```

Comment:
This could be useful in other places as well. It does not belong to this class when you consider Single Responsibility Principle then Membership Manager has nothing to do with converting exceptions to array. Here's what I suggest:

1. Create new interface
```php
namespace Mautic\CoreBundle\Exception;

abstract class FlattenableException extends Exception
{
    /**
     * @return array{'message': string, 'file': string, 'line': int, 'trace': string}
     */
    public function toArray(): array
    {
        return [
            'message' => $this->getMessage(),
            'file'    => $this->getFile(),
            'line'    => $this->getLine(),
            'trace'   => $this->getTraceAsString(),
        ];
    }
}
```
_(not sure if this class can be abstract, but give it a try)_
Then instead of the exceptions that are caught here like `ContactCannotBeAddedToCampaignException` would extend just `\Exception` they will extend `FlattenableException` and then you'll be able to call `$exception->toArray()` on them directly.

Perks:
1. The responsibility is in the right place
2. Any other exceptions can easily use it and become flattenable